public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -97,6 +111,69 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
 
   private forceRouteReloadCounter = 0;
   state: State = this.getCleanState();
+  maxFileSize = 500000;
+
+  onFileDrop = (acceptedFiles: File[], fileRejections: FileRejection[], event: DropEvent) => {
+    console.log(acceptedFiles);
+    acceptedFiles.forEach((file) => {
+      const reader = new FileReader();
+      reader.readAsArrayBuffer(file);
+      reader.onload = () => {
+        const result = reader.result;
+        if (result && result instanceof ArrayBuffer) {
+          const dataFrames = readSpreadsheet(result);
+          const snapshot: DataFrameJSON[] = [];
+          dataFrames.forEach((df) => {
+            const dataframeJson = dataFrameToJSON(df);
+            snapshot.push(dataframeJson);
+          });
+          const grafanaDS = {
+            type: 'grafana',
+            uid: 'grafana',
+          };
+          this.props.dashboard?.addPanel({
+            type: 'table',
+            gridPos: { x: 0, y: 0, w: 12, h: 8 },
+            title: file.name,
+            datasource: grafanaDS,
+            targets: [
+              {
+                queryType: GrafanaQueryType.Snapshot,
+                snapshot,
+                datasource: grafanaDS,
+              },
+            ],
+          });
+        }
+      };
+    });
+
+    fileRejections.forEach((fileRejection) => {
+      const errors = fileRejection.errors.map((error) => {
+        switch (error.code) {
+          case ErrorCode.FileTooLarge:
+            const formattedSize = getValueFormat('decbytes')(this.maxFileSize);
+            return `File is larger than ${formattedValueToString(formattedSize)}.`;
+          case ErrorCode.FileInvalidType:
+            return 'File must be a spreadsheet.';
```

Comment:
Would be good to have more guidance in error message such as having a short list of file types. Right now it's hard to tell what exactly will work and what won't

Reply:
Addressed here: https://github.com/grafana/grafana/pull/62688/commits/d38cec40182393edc3b57ca565e5a4069c523c9f#diff-a09c348290db927b350a7af939ca2bca0db0d3151f1999a963d1bf8356fa9c5eR149